horizontal_rule_test.py

```diff
@@ -1,6 +1,54 @@
+import pytest
 from horizontal_rule import HorizontalRule
 
 
 def test_render():
     horiz_rule = HorizontalRule('')
     assert horiz_rule.render() == '<hr>'
+
+
+@pytest.mark.parametrize(
+    "content, start, end",
+    [
+        ('***', 0, 3),
+        ('____', 0, 4),
+        ('-----', 0, 5),
+        (
+            '''---
+            abc
+            ''', 0, 3
+        ),
+        (
+            '''abc
+*****
+            ''', 4, 9
+        ),
+    ]
+)
+def test_parse(content, start, end):
+    (start_index, end_index, hori_rule) = HorizontalRule.parse(content)
+    assert start_index == start
+    assert end_index == end
+    assert hori_rule._content == ''
+
+
+@pytest.mark.parametrize(
+    "content",
+    [
+        # not match
+        ('**'),
+        ('*_-'),
+        ('\---'),
+        ('a___'),
+        ('*****a'),
+        ('***a**'),
+        (' ------'),
+        ('______ '),
```

Comment:
shouldn't this one be a valid case?

Reply:
space after

Reply:
nvm, GitHub supports space. will update it

Reply:
make sure "Resolve conversation" after resolve it.